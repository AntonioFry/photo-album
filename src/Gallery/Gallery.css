.gallery-section {
  display: grid;
  grid-template-columns: 1fr 1fr;
  justify-content: space-evenly;
  padding: 30px;
}