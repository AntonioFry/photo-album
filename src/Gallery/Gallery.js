import React from 'react';
import PhotoCard from '../PhotoCard/PhotoCard';
import { imageData } from '../Data/Data';
import './Gallery.css';

const Gallery = () => {
  const photoCards = imageData.map(imgData => {
    return (
      <PhotoCard 
        {...imgData}
      />
    )
  })
  return (
    <section className="gallery-section">
      {photoCards}
    </section>
  )
}

export default Gallery;