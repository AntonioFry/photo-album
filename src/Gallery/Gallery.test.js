import React from 'react';
import ReactDOM from 'react-dom';
import Gallery from './Gallery';
import renderer from 'react-test-renderer';

import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';

describe('Gallery', () => {

  afterEach(cleanup);

  it('Should render without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Gallery />, div);
  });

  it('Should render correctly', () => {
    const { getByTestId } = render(<Gallery />);
    expect(getByTestId('photo-comment-1')).toHaveTextContent('This is one the first image I took of any sort of insect. After being able to see how much detail that was visible in the image versus the naked eye, I decided to try taking more of the small critters that roam the earth.')
  });

  it('Should match snapshot', () => {
    const tree = renderer.create(<Gallery />);
    expect(tree).toMatchSnapshot();
  });

});