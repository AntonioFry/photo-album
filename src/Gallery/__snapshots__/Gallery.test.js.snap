// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery Should match snapshot 1`] = `
<section
  className="gallery-section"
>
  <article
    className="photo-card"
    style={
      Object {
        "backgroundImage": "url(ants-and-plants.JPG)",
        "backgroundSize": "cover",
      }
    }
  >
    <div
      className="photo-comment-container"
    >
      <p
        className="photo-comment"
        data-testid="photo-comment-1"
      >
        This is one the first image I took of any sort of insect. After being able to see how much detail that was visible in the image versus the naked eye, I decided to try taking more of the small critters that roam the earth.
      </p>
    </div>
  </article>
  <article
    className="photo-card"
    style={
      Object {
        "backgroundImage": "url(moon.JPG)",
        "backgroundSize": "cover",
      }
    }
  >
    <div
      className="photo-comment-container"
    >
      <p
        className="photo-comment"
        data-testid="photo-comment-2"
      >
        I have always thougth it's always been cool to see the Moon during the day, and so I took a picture. It maybe common photo to find on the web, but it's still cool to see the craters so clearly when it's from a photo you took.
      </p>
    </div>
  </article>
  <article
    className="photo-card"
    style={
      Object {
        "backgroundImage": "url(honeybee.JPG)",
        "backgroundSize": "cover",
      }
    }
  >
    <div
      className="photo-comment-container"
    >
      <p
        className="photo-comment"
        data-testid="photo-comment-3"
      >
        Though this photo isn't my favorite, it was still a lucky shot to catch a bee polinating
      </p>
    </div>
  </article>
  <article
    className="photo-card"
    style={
      Object {
        "backgroundImage": "url(wasp.JPG)",
        "backgroundSize": "cover",
      }
    }
  >
    <div
      className="photo-comment-container"
    >
      <p
        className="photo-comment"
        data-testid="photo-comment-4"
      >
        Took this photo from across the room using a longer lens. I was amazed to also see how much detail was able to be captured. I was also able to free the bugger from my house.
      </p>
    </div>
  </article>
  <article
    className="photo-card"
    style={
      Object {
        "backgroundImage": "url(jewell.JPG)",
        "backgroundSize": "cover",
      }
    }
  >
    <div
      className="photo-comment-container"
    >
      <p
        className="photo-comment"
        data-testid="photo-comment-5"
      >
        This is nice shot of a downhill street in my neighborhood that faces the mountains. However I would say it looks a lot nicer when actually there.
      </p>
    </div>
  </article>
</section>
`;
